src/providers/saml-jackson.ts

```diff
@@ -0,0 +1,51 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface SAMLJacksonProfile {
+  id: string
+  email: string
+  firstName: string
+  lastName: string
+  name: string
+  email_verified: boolean
+  image: null
+}
+
+export default function SAMLJackson<
+  P extends Record<string, any> = SAMLJacksonProfile
+>(options: OAuthUserConfig<P>): OAuthConfig<P> {
+  return {
+    id: "saml-jackson",
+    name: "BoxyHQ SAML Jackson",
+    type: "oauth",
+    version: "2.0",
+    checks: ["pkce", "state"],
+    authorization: {
+      url: `${options.issuer}/api/oauth/authorize`,
+      params: {
+        scope: "",
+        response_type: "code",
+        provider: "saml",
+      },
+    },
+    token: {
+      url: `${options.issuer}/api/oauth/token`,
+      params: { grant_type: "authorization_code" },
+    },
+    userinfo: `${options.issuer}/api/oauth/userinfo`,
+    profile(profile) {
+      return {
+        id: profile.id || "",
+        firstName: profile.firstName || "",
+        lastName: profile.lastName || "",
+        email: profile.email || "",
+        name: `${profile.firstName || ""} ${profile.lastName || ""}`.trim(),
+        email_verified: true,
+        image: null,
+      }
+    },
+    options: {
+      clientId: options.clientId || "dummy",
+      clientSecret: options.clientSecret || "dummy",
+    },
```

Comment:
```suggestion
    options
```

This should be kept as-is, so we can merge the user's overrides on top of the defaults from above.

If "dummy" is a special client, I recommend adding this information to the docs instead (https://github.com/nextauthjs/docs/pull/217), to mention that users can experiment with this provider using the `"dummy"` client id and secret values.